### PR TITLE
autoencoder example and learning rate multipler for optimizer

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

**Comment by tqchen** [comment]
The lr_mult is a bit debatable. A better way might be setup per index lr into optimizer. Can you remove the lr_mult for now. The rest of example looks good to me. Please add a README.md to guide user on   how to run examples.

**Comment by piiswrong** [comment]
the example depends on it. we can merge this after we have a solution

**Comment by tqchen** [comment]
As discussed, let us consider move the lr scaling to the optimizer

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def create_optimizer(name, rescale_grad=1, **kwargs):
     def __init__(self, rescale_grad=1):
         self.epoch = 0
         self.rescale_grad = rescale_grad
+        self.lr_mult = {}
```
Comment (tqchen) [inline python/mxnet/optimizer.py]: consider rename to lr_scale?

**Comment by tqchen** [comment]
I think this PR is good to go. Just suggest change lr_mult to ```lr_scale```, which seems to be more clear

**Comment by tqchen** [comment]
Also convention issue in python3, dict.get should not use keyword argument, but simply get(index, 1.0)

**Pull request merged**